lib/catalog/service_plan_fields.rb

```diff
@@ -0,0 +1,35 @@
+module Catalog
+  class ServicePlanFields
+    attr_reader :fields
+
+    def initialize(order_item)
+      @order_item = order_item
+    end
+
+    def process
+      @fields = Array(service_plan_schema.with_indifferent_access.dig(:schema, :fields))
```

Comment:
The `dig(:schema, :fields)` already returns type of Array, right?

Reply:
I see, this will return `[]` when `fields` is `nil` 👍

Reply:
Array() will turn nil to [], just in case `dig` cannot dig anything out.